src/plugins/workspace/public/components/workspace_creator/workspace_creator.test.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { WorkspaceCreator } from './workspace_creator';
+import { PublicAppInfo } from 'opensearch-dashboards/public';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import * as opensearchReactExports from '../../../../opensearch_dashboards_react/public';
+import { BehaviorSubject } from 'rxjs';
+import { WorkspaceFormSubmitData } from './workspace_form';
+
+jest.mock('../../../../opensearch_dashboards_react/public', () => ({
+  ...jest.requireActual('../../../../opensearch_dashboards_react/public'),
+  __esModule: true,
+}));
+
+const workspaceClientCreate = jest
+  .fn()
+  .mockReturnValue({ result: { id: 'successResult' }, success: true });
+
+const navigateToApp = jest.fn();
+const notificationToastsAddSuccess = jest.fn();
+const notificationToastsAddDanger = jest.fn();
+const PublicAPPInfoMap = new Map([
+  ['app1', { id: 'app1', title: 'app1' }],
+  ['app2', { id: 'app2', title: 'app2', category: { id: 'category1', label: 'category1' } }],
+  ['app3', { id: 'app3', category: { id: 'category1', label: 'category1' } }],
+  ['app4', { id: 'app4', category: { id: 'category2', label: 'category2' } }],
+  ['app5', { id: 'app5', category: { id: 'category2', label: 'category2' } }],
+]);
+
+jest.spyOn(opensearchReactExports, 'useOpenSearchDashboards').mockReturnValue({
+  services: {
+    application: {
+      navigateToApp,
+      getUrlForApp: jest.fn(),
+      applications$: new BehaviorSubject<Map<string, PublicAppInfo>>(PublicAPPInfoMap as any),
+    },
+    http: {
+      basePath: {
+        remove: jest.fn(),
+        prepend: jest.fn(),
+      },
+    },
+    notifications: {
+      toasts: {
+        addDanger: notificationToastsAddDanger,
+        addSuccess: notificationToastsAddSuccess,
+      },
+    },
+    workspaceClient: {
+      create: workspaceClientCreate,
+    },
+  },
```

Comment:
can we use 
```
...coreServiceMock.create()
...{
   application: {...},
   notifications: {...},
   workspaceClient: {...}
}
```

In this case, if coreStart adds any new fields, we do not need to modify the mock here.